### PR TITLE
Skip version check for cloud v2 rightnow

### PR DESCRIPTION
**Pull request by wanliAlex**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
we skip the version check for cloud v2

* **What is the current behavior?** (You can also link to an open issue here)
we have the version check for all urls

* **What is the new behavior (if this is a feature change)?**
we skip version check for specific urls

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no

* **Other information**:
no

### CODE REVIEW

File: src/marqo/client.py [hunk 1]
```diff
@@ -199,12 +200,34 @@ def get_cpu_info(self):
...
+        try:
+            if self.url not in marqo_url_and_version_cache:
+                marqo_url_and_version_cache[self.url] = self.get_marqo()["version"]
+            marqo_version = marqo_url_and_version_cache[self.url]
+            if versioning_helpers.parse(marqo_version) < versioning_helpers.parse(min_ver):
+                mq_logger.warning(f"Your Marqo Python client requires a minimum Marqo version of "
+                                  f"{minimum_supported_marqo_version()} to function properly, while your Marqo version is {marqo_version}. "
+                                  f"Please upgrade your Marqo instance to avoid potential errors. "
+                                  f"If you have already upgraded your Marqo instance but still having this warning, please import you Marqo Python client again and retry.")
+        except JSONDecodeError as e:
```
Comment (vicilliar) [inline src/marqo/client.py]: In what situation do we see a `JSONDecodeError` happening in the version check?

Comment (wanliAlex) [reply src/marqo/client.py]: When it reaches the cloud v2 API and it is not in the skip list. One example is “https://api.marqo.ai/api/”

File: src/marqo/client.py [hunk 2]
```diff
@@ -199,12 +202,35 @@ def get_cpu_info(self):
...
+            if versioning_helpers.parse(marqo_version) < versioning_helpers.parse(min_ver):
+                mq_logger.warning(f"Your Marqo Python client requires a minimum Marqo version of "
+                                  f"{minimum_supported_marqo_version()} to function properly, but your Marqo version is {marqo_version}. "
+                                  f"Please upgrade your Marqo instance to avoid potential errors. "
+                                  f"If you have already changed your Marqo instance but still get this warning, please restart your Marqo client Python interpreter.")
+        except (JSONDecodeError, BackendTimeoutError, BackendCommunicationError,requests.exceptions.RequestException) as e:
+            mq_logger.warning(
+                f"Marqo encountered a problem trying to check the Marqo version found at `{self.url}`. "
+                f"The minimum supported Marqo version for this client is {min_ver}. "
+                f"If you are sure the Marqo version is compatible with this client, you can ignore this message. "
```
Comment (farshidz) [inline src/marqo/client.py]: If you are sure **your** Marqo version

File: src/marqo/client.py [hunk 3]
```diff
@@ -199,12 +202,35 @@ def get_cpu_info(self):
...
-        if versioning_helpers.parse(marqo_version) < versioning_helpers.parse(minimum_supported_marqo_version()):
-            mq_logger.warning(f"Your Marqo Python client requires a minimum Marqo version of "
-                              f"{minimum_supported_marqo_version()} to function properly, while your Marqo version is {marqo_version}. "
-                              f"Please upgrade your Marqo instance to avoid potential errors. "
-                              f"If you have already upgraded your Marqo instance but still having this warning, please import you Marqo Python client again and retry.")
+        min_ver = minimum_supported_marqo_version()
+
+        # Skip the check if the url is previously labelled as "_skipped"
+        if self.url in marqo_url_and_version_cache and marqo_url_and_version_cache[self.url] == "_skipped":
+            return
```
Comment (farshidz) [inline src/marqo/client.py:214]: We should log the same warning here as well

File: src/marqo/client.py [hunk 4]
```diff
@@ -199,12 +202,35 @@ def get_cpu_info(self):
...
+
+        # Skip the check for Marqo CloudV2 APIs right now
+        skip_version_check_url = ["https://api.marqo.ai", "https://cloud.marqo.ai"]
+        if self.url in skip_version_check_url:
+            marqo_url_and_version_cache[self.url] = "_skipped"
+            return
```
Comment (farshidz) [inline src/marqo/client.py:221]: We should log the same warning here as well

**Review by farshidz — CHANGES_REQUESTED** [review]

File: src/marqo/client.py [hunk 5]
```diff
@@ -199,12 +202,35 @@ def get_cpu_info(self):
...
+        try:
+            if self.url not in marqo_url_and_version_cache:
+                marqo_url_and_version_cache[self.url] = self.get_marqo()["version"]
+            marqo_version = marqo_url_and_version_cache[self.url]
+            if versioning_helpers.parse(marqo_version) < versioning_helpers.parse(min_ver):
+                mq_logger.warning(f"Your Marqo Python client requires a minimum Marqo version of "
+                                  f"{minimum_supported_marqo_version()} to function properly, but your Marqo version is {marqo_version}. "
+                                  f"Please upgrade your Marqo instance to avoid potential errors. "
+                                  f"If you have already changed your Marqo instance but still get this warning, please restart your Marqo client Python interpreter.")
+        except (JSONDecodeError, BackendTimeoutError, BackendCommunicationError,requests.exceptions.RequestException) as e:
```
Comment (farshidz) [inline src/marqo/client.py]: As discussed, let's catch MarqoWebError, RequestException , TypeError and KeyError

**Review by farshidz — CHANGES_REQUESTED** [review]

File: src/marqo/client.py (hunk 5 above)
Comment (farshidz) [reply src/marqo/client.py]: It would be good to have some tests for these exception scenarios as well

File: src/marqo/client.py [hunk 6]
```diff
@@ -199,12 +202,37 @@ def get_cpu_info(self):
...
+        try:
+            if self.url not in marqo_url_and_version_cache:
+                marqo_url_and_version_cache[self.url] = self.get_marqo()["version"]
+            marqo_version = marqo_url_and_version_cache[self.url]
+            if versioning_helpers.parse(marqo_version) < versioning_helpers.parse(min_ver):
+                mq_logger.warning(f"Your Marqo Python client requires a minimum Marqo version of "
+                                  f"{minimum_supported_marqo_version()} to function properly, but your Marqo version is {marqo_version}. "
+                                  f"Please upgrade your Marqo instance to avoid potential errors. "
+                                  f"If you have already changed your Marqo instance but still get this warning, please restart your Marqo client Python interpreter.")
+        except (MarqoWebError, RequestException, TypeError, KeyError) as e:
```
Comment (farshidz) [inline src/marqo/client.py:233]: As discussed please also include `requests.exceptions.Timeout`

File: tests/v0_tests/test_client.py [hunk 7]
```diff
@@ -54,14 +58,58 @@ def test_version_check_instantiation(self):
...
+                # Get the warning message
+                warning_message = mock_warning.call_args[0][0]
+
+                # Assert the message is what you expect
+                self.assertIn("Marqo encountered a problem trying to check the Marqo version found", warning_message)
+                self.assertEqual(marqo_url_and_version_cache, dict({self.client_settings["url"] : "_skipped"}))
+
+                marqo_url_and_version_cache.clear()
+
     def test_version_check_multiple_instantiation(self):
```
Comment (farshidz) [inline tests/v0_tests/test_client.py:104]: Can we have one test similar to this but for skipped scenario, where we make sure multiple instantiations make only one `get_marqo` attempt and also log a warning?

**Review by farshidz — CHANGES_REQUESTED** [review]

**Review by farshidz — APPROVED** [review]

**Pull request merged**